src/controllers/reset-password.js

```
@@ -0,0 +1,83 @@
+/* eslint-disable consistent-return */
+const { User } = require('models');
+const Joi = require('joi');
+const { scryptSync } = require('crypto');
+const jwt = require('jsonwebtoken');
+const { cryptoLink, decryptLink } = require('./crypto');
+// eslint-disable-next-line no-useless-escape
+const passwordRegex = /^(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.*[!@#\$%\^&\*])(?=.{8,})/;
+const setResetPassword = async (req, res) => {
+  try {
+    const schema = Joi.object({ email: Joi.string().email().required() });
+    const { error } = schema.validate(req.body);
+    if (error) return res.status(201).send(error.details[0].message);
```

we should return error status code if we have errors (maybe 400),  201 is a success status code
also, you can throw any error from `src/lib/errors.js` and it will be handled by the global handler